src/main/java/common/DriverBinariesSetter.java

```diff
@@ -1,40 +1,80 @@
 
     private DriverBinariesSetter() {
     }
 
+    public static WebDriver driver;
+
     private static final String EDGE = "edge";
     private static final String CHROME = "chrome";
     private static final String FIREFOX = "firefox";
-    private static final String INTERNET_EXPLORER = "iexplorer";
-    private static final String PROVIDED = "provided";
-    private static final String PROVIDED_TYPE = System.getProperty("webdriver.provided.type", CHROME);
+    private static final String SAUCE_LAB = "sauceLab";
+
 
-    public static void setBinariesForUsedWebDriver(String browser) {
+    public static void setBinariesForUsedWebDriver(String browser) throws MalformedURLException {
         //Java 14 switch
         switch (browser) {
-            case INTERNET_EXPLORER -> setBinaryByName(INTERNET_EXPLORER);
             case EDGE -> setBinaryByName(EDGE);
-            case PROVIDED -> setBinaryByName(PROVIDED_TYPE);
+            case FIREFOX -> setBinaryByName(FIREFOX);
+            case SAUCE_LAB -> setBinaryByName(SAUCE_LAB);
             default -> setBinaryByName(CHROME);
         }
     }
 
-    private static void setBinaryByName(String name) {
+    private static void setBinaryByName(String name) throws MalformedURLException {
         //Java 14 switch
         switch (name) {
-            case FIREFOX -> WebDriverManager.firefoxdriver().setup();
-            case EDGE -> WebDriverManager.edgedriver()
-                    .driverVersion(System.getProperty("edge.driver.version",
-                            StringUtils.EMPTY)).setup();
-            default -> WebDriverManager.chromedriver()
-                    .driverVersion(System.getProperty("chrome.driver.version",
-                            StringUtils.EMPTY)).setup();
+            case FIREFOX -> {
+                WebDriverManager.firefoxdriver().setup();
+                driver = new FirefoxDriver();
+            }
+            case EDGE -> {
+                WebDriverManager.edgedriver()
+                        .driverVersion(System.getProperty("edge.driver.version",
+                                StringUtils.EMPTY)).setup();
+                driver = new EdgeDriver();
+            }
+            case SAUCE_LAB -> callSauceLabProperties();
+            default -> {
+                WebDriverManager.chromedriver()
+                        .driverVersion(System.getProperty("chrome.driver.version",
+                                StringUtils.EMPTY)).setup();
+                driver = new ChromeDriver();
+            }
         }
+        WebDriverRunner.setWebDriver(driver);
+    }
+
+    private static void callSauceLabProperties() throws MalformedURLException {
+        URL url = new URL("https://oauth-kulyasaleksandra-cb8f8:5805de78-2c54-423a-bf5e-bbd008467f6f@ondemand" +
+                ".eu-central-1.saucelabs.com:443/wd/hub");
+        DesiredCapabilities desiredCapabilities = new DesiredCapabilities();
+        ChromeOptions chromeOptions = new ChromeOptions();
```

Comment:
this method implementation works only with Chrome. Sure you can move "platform" and "browser" to properties but what about other browsers? In this case, you need to make changes here or add another method that breaks the SOLID "opened closed principle". Let's discuss